modules/gateways/blockonomics.php

```diff
@@ -84,11 +83,43 @@ function reqListener () {
 				newBtn.disabled = false;
 			}
 
+			function reqListener () {
+				responseObj = {};
+				switch (this.status) {
+					case 200:
+						responseObj = JSON.parse(this.responseText);
+						break;
+					case 401:
+						responseObj.error = true;
+						responseObj.errorStr = 'There was an issue accessing the test file. Please make sure all plugin files have correct permissions.'
+						break;
+					case 404:
+						responseObj.error = true;
+						responseObj.errorStr = 'There was an issue finding the test file. Please make sure all plugin files are copied correctly.'
+						break;
+					default:
+						responseObj.error = true;
+						responseObj.errorStr = 'There was an issue when requesting testSetup.php';
+						break;
+				}
+				showTestResult(responseObj);
+			}
+
+			function timeoutListener () {
+				responseObj = {};
+				responseObj.error = true;
+				responseObj.errorStr = "There was an issue accessing the test file. Please make sure all plugin files have correct \
+																	permissions and that they are copied correctly"
+				showTestResult(responseObj);
+			}
+
 			newBtn.onclick = function() {
 				var testSetupUrl = "$system_url" + "testSetup.php";
 				var oReq = new XMLHttpRequest();
 				oReq.addEventListener("load", reqListener);
 				oReq.open("GET", testSetupUrl);
+				oReq.timeout = 10000; // 10 seconds
```

Comment:
Try to reproduce timeout
- If its doesn't reproduce. remove timeout
- If it reproduces, set same message as reqListener